controllers/azuremanagedcontrolplane_reconciler.go

```diff
@@ -156,9 +157,32 @@ func (r *azureManagedControlPlaneService) reconcileKubeconfig(ctx context.Contex
 			}
 			return nil
 		}); err != nil {
-			return errors.Wrap(err, "failed to kubeconfig secret for cluster")
+			return errors.Wrap(err, "failed to reconcile kubeconfig secret for cluster")
 		}
 	}
 
+	// store cluster-info for the cluster with the admin kubeconfig.
+	kubeconfigFile, err := clientcmd.Load(kubeConfigs[0])
+	if err != nil {
+		return errors.Wrap(err, "failed to turn aks credentials into kubeconfig file struct")
+	}
+
+	cluster := kubeconfigFile.Contexts[kubeconfigFile.CurrentContext].Cluster
+	caData := kubeconfigFile.Clusters[cluster].CertificateAuthorityData
+	caSecret := r.scope.MakeClusterCA()
+	if _, err := controllerutil.CreateOrUpdate(ctx, r.kubeclient, caSecret, func() error {
```

Comment:
It's to make the KubeadmConfig controller happy when it generates the bootstrap data for the machinepool, it would normally be created when reconciling the KCP object but in this case there is none

it follows the convention defined here https://cluster-api.sigs.k8s.io/tasks/certs/using-custom-certificates